compiler/luci/pass/src/helpers/LayerInfoMap.cpp

```diff
@@ -22,6 +22,154 @@
+      return loco::must_cast<luci::CircleNode *>(unpack_out->input());
+    }
+    // TODO: Support these ops
+    case luci::CircleOpcode::CIRCLEBIDIRECTIONAL_SEQUENCE_LSTM_OUT:
+    case luci::CircleOpcode::CIRCLECUSTOMOUT:
+    case luci::CircleOpcode::CIRCLEIFOUT:
+    case luci::CircleOpcode::CIRCLENONMAXSUPPRESSIONV4OUT:
+    case luci::CircleOpcode::CIRCLENONMAXSUPPRESSIONV5OUT:
+    case luci::CircleOpcode::CIRCLEWHILEOUT:
+      throw std::runtime_error("Unsupported op now");
+    default:
+      return nullptr;
+  }
+}
+
+bool same_setting(const LayerInfo &left, const LayerInfo &right)
+{
+  return left.dtype == right.dtype and left.granularity == right.granularity;
+}
+
+void add_multi_output_node(LayerInfoMap &info_by_name, LayerInfo &layer_info,
+                           const luci::CircleNode *node, std::vector<LayerInfo> &layers_info)
+{
+  assert(is_multiple_output_node(node)); // FIX_CALLER_UNLESS
+
+  const auto succs_nodes = loco::succs(node);
+  const auto name = node->name();
+
+  if (info_by_name.find(name) != info_by_name.end())
+  {
+    // Check that all outputs have equal dtype and granularity
+    for (const auto succs_node : succs_nodes)
+    {
+      const auto succs_circle_node = loco::must_cast<luci::CircleNode *>(succs_node);
+
+      const auto it = info_by_name.find(succs_circle_node->name());
+      if (it != info_by_name.end() and not same_setting(layer_info, (it->second)))
+        throw std::runtime_error("Outputs of multiple-output nodes should have equal dtype and "
+                                 "granularity. Check the quantization configuration file");
+    }
+    return;
+  }
+
+  // Add multiple output node to info_by_name
+  if (layer_info.name == name)
+  {
+    info_by_name[name] = layer_info;
+  }
+  else
+  {
+    auto it = std::find_if(layers_info.begin(), layers_info.end(),
+                           [&name](const auto &info) { return name == info.name; });
+
+    if (it == layers_info.end())
+    {
+      info_by_name[name] = {name, layer_info.dtype, layer_info.granularity};
+    }
+    else
+    {
+      info_by_name[name] = *it;
+    }
+  }
+
+  // Add outputs node to info_by_name
+  for (const auto succs_node : succs_nodes)
+  {
+    const auto succs_circle_node = loco::must_cast<luci::CircleNode *>(succs_node);
+    const auto succs_circle_node_name = succs_circle_node->name();
+    if (name != succs_circle_node_name)
+    {
+      auto it = std::find_if(layers_info.begin(), layers_info.end(),
+                             [&succs_circle_node_name](const auto &info) {
+                               return succs_circle_node_name == info.name;
+                             });
+
+      if (it == layers_info.end())
+      {
+        info_by_name[succs_circle_node_name] = {succs_circle_node_name, layer_info.dtype,
+                                                layer_info.granularity};
+      }
+      else
+      {
+        info_by_name[succs_circle_node_name] = *it;
+      }
+    }
```

Comment:
Similarly, is it possible to simplify the code?

```suggestion
    info_by_name[succs_circle_node_name] = {succs_circle_node_name, layer_info.dtype,
                                                layer_info.granularity};
```